src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/NearestPointFuseRegularGrids.cpp

```diff
@@ -0,0 +1,198 @@
+        if((x - sampleOrigin[0]) < 0)
+        {
+          continue;
+        }
+        if((y - sampleOrigin[1]) < 0)
+        {
+          continue;
+        }
+        if((z - sampleOrigin[2]) < 0)
+        {
+          continue;
+        }
+
+        col = usize((x - sampleOrigin[0]) / sampleRes[0]);
+        row = usize((y - sampleOrigin[1]) / sampleRes[1]);
+        plane = usize((z - sampleOrigin[2]) / sampleRes[2]);
+        if(col >= sampleDims[0] || row >= sampleDims[1] || plane >= sampleDims[2])
+        {
+          continue;
+        }
+
+        sampleIndex = (plane * sampleDims[0] * sampleDims[1]) + (row * sampleDims[0]) + col;
+        refIndex = planeComp + rowComp + k;
+
+        usize numComps = destArray.getNumberOfComponents();
+        usize sourceIndex = sampleIndex * inputArray.getNumberOfComponents();
+        usize destIndex = refIndex * numComps;
+        for(usize offset = 0; offset < numComps; ++offset)
+        {
+          destArray[destIndex + offset] = inputArray.at(sourceIndex + offset);
+        }
+      }
+    }
+  }
+}
+
+template <typename T>
+class CopyFromIndicesListImpl
+{
+public:
+  CopyFromIndicesListImpl(const IArray& sourceArray, IArray& destArray, const ImageGeom& sampleImageGeom, const ImageGeom& refImageGeom, float64 fillValue, const std::atomic_bool& shouldCancel)
+  : m_SourceArray(sourceArray)
+  , m_DestArray(destArray)
+  , m_ArrayType(destArray.getArrayType())
+  , m_SampleGeom(sampleImageGeom)
+  , m_RefGeom(refImageGeom)
+  , m_FillValue(fillValue)
+  , m_ShouldCancel(shouldCancel)
+  {
+  }
+  ~CopyFromIndicesListImpl() = default;
+
+  CopyFromIndicesListImpl(const CopyFromIndicesListImpl&) = default;
+  CopyFromIndicesListImpl(CopyFromIndicesListImpl&&) noexcept = default;
+  CopyFromIndicesListImpl& operator=(const CopyFromIndicesListImpl&) = delete;
+  CopyFromIndicesListImpl& operator=(CopyFromIndicesListImpl&&) noexcept = delete;
+
+  void operator()() const
+  {
+    if(m_ArrayType == IArray::ArrayType::NeighborListArray)
+    {
+      return;
+      //      using NeighborListT = NeighborList<T>;
+      //      auto& destArray = dynamic_cast<NeighborListT&>(m_DestArray);
+      //      // Make sure the destination array is allocated AND each tuple list is initialized, so we can use the [] operator to copy over the data
+      //      if(destArray.getValues().empty() || destArray.getList(0) == nullptr)
+      //      {
+      //        destArray.addEntry(destArray.getNumberOfTuples() - 1, 0);
+      //      }
+      //      CopyData<NeighborListT>(dynamic_cast<const NeighborListT&>(m_SourceArray1), destArray);
+    }
+    else if(m_ArrayType == IArray::ArrayType::DataArray)
+    {
+      using DataArrayT = DataArray<T>;
+      auto destArray = dynamic_cast<DataArrayT&>(m_DestArray);
+      destArray.fill(static_cast<T>(m_FillValue));
+      CopyData<DataArrayT>(dynamic_cast<const DataArrayT&>(m_SourceArray), destArray, m_SampleGeom, m_RefGeom, m_ShouldCancel);
```

Comment:
@mmarineBlueQuartz the issue is that the entire array must be filled because its not a straight forward copy. One possible outcome could be structured as follows:

Source Array: { 1, 2, 3, 4, 5}
Dest Array: { 0, 0, 0, 0, 0, 0, 0}

With your suggestion one would use your described fill to get the following assuming the fill value is 10:

Source Array: { 1, 2, 3, 4, 5}
Dest Array: { 0, 0, 0, 0, 0, 10, 10}

Leaving the remaining spaces to be filled. This is valid, however the end goal of the filter is to only copy over the overlap between the two arrays based on origin and spacing. So one possible solution could be:

Dest Array: {10, 10, 10, 10, 3, 10, 5}

To fix this I substituted by moving the fill in front of every continue in the copyData function. I am pushing the change now.